podpac/core/node.py

```diff
@@ -1208,33 +1208,97 @@ def get_ui_spec(cls, help_as_html=False):
+            module_root = d.get("plugin", "podpac")
+            node_string = "%s.%s" % (module_root, d["node"])
+            module_name, node_name = node_string.rsplit(".", 1)
+            try:
+                module = importlib.import_module(module_name)
+            except ImportError:
+                raise ValueError("Invalid definition for node '%s': no module found '%s'" % (name, module_name))
+            try:
+                node_class = getattr(module, node_name)
+            except AttributeError:
+                raise ValueError(
+                    "Invalid definition for node '%s': class '%s' not found in module '%s'"
+                    % (found_name, node_name, module_name)
+                )
+
+            kwargs = {}
+            for k, v in d.get("attrs", {}).items():
+                kwargs[k] = v
+
+            for k, v in d.get("inputs", {}).items():
+                kwargs[k], nodes = _lookup_input(nodes, found_name, v, definition)
+
+            for k, v in d.get("lookup_attrs", {}).items():
+                kwargs[k] = _lookup_attr(nodes, found_name, v)
+
+            if "style" in d:
+                style_class = getattr(node_class, "style", Style)
+                if isinstance(style_class, tl.TraitType):
+                    # Now we actually have to look through the class to see
+                    # if there is a custom initializer for style
+                    for attr in dir(node_class):
+                        atr = getattr(node_class, attr)
+                        if not isinstance(atr, tl.traitlets.DefaultHandler) or atr.trait_name != "style":
+                            continue
+                        try:
+                            style_class = atr(node_class)
+                        except Exception as e:
+                            # print ("couldn't make style from class", e)
+                            try:
+                                style_class = atr(node_class())
+                            except:
+                                # print ("couldn't make style from class instance", e)
+                                style_class = style_class.klass
+                try:
+                    kwargs["style"] = style_class.from_definition(d["style"])
+                except Exception as e:
+                    kwargs["style"] = Style.from_definition(d["style"])
+                    # print ("couldn't make style from inferred style class", e)
+
+            for k in d:
+                if k not in ["node", "inputs", "attrs", "lookup_attrs", "plugin", "style"]:
+                    raise ValueError("Invalid definition for node '%s': unexpected property '%s'" % (name, k))
+
+            nodes[found_name] = node_class(**kwargs)
+            # found!
+            break
 
     if not value in nodes:
         raise ValueError(
             "Invalid definition for node '%s': reference to nonexistent node '%s' in inputs" % (name, value)
         )
-
+    print(value)
```

Comment:
Errant print statement

Reply:
Deleted in 3fb5048ef08d00fd69e95d05343ebe02db73fc98